src/ring_theory/algebra_tower.lean

```diff
@@ -458,3 +458,52 @@ by exactI fg_of_injective (is_scalar_tower.to_alg_hom B₀ B C).to_linear_map
   (linear_map.ker_eq_bot.2 hBCi)
 
 end artin_tate
+
+section alg_hom_tower
+
+variables {A} {C D : Type*} [comm_semiring A]  [comm_semiring C] [comm_semiring D]
+  [algebra A C] [algebra A D]
+
+variables (f : C →ₐ[A] D) (B) [comm_semiring B] [algebra A B] [algebra B C] [is_scalar_tower A B C]
+
+/-- Restrict the domain of an alg_hom -/
+def alg_hom.restrict : B →ₐ[A] D := f.comp (is_scalar_tower.to_alg_hom A B C)
+
+/-- Extend the scalars of an alg_hom -/
+def alg_hom.extend_base : @alg_hom B C D _ _ _ _ (f.restrict B).to_ring_hom.to_algebra :=
+{ commutes' := λ _, rfl .. f }
+
+variables {B}
+
+/-- Combine two alg_hom's that are in a tower -/
+def alg_hom_compose (f : B →ₐ[A] D) (g : @alg_hom B C D _ _ _ _ f.to_ring_hom.to_algebra) :
+  C →ₐ[A] D :=
+{ to_fun := g,
```

Comment:
This name feels a bit too close to `alg_hom.comp` to me, especially for something that doesn't even do a composition!

Perhaps this should be in the `is_scalar_tower` namespace?

Reply:
Sure. Are you suggesting to rename it to `is_scalar_tower.alg_hom_compose`?

Reply:
Maybe. I'm more just flagging this for somehow who knows more about scalar towers than me to think of a better name :)

Reply:
This def turned out to be a redundant version of `is_scalar_tower.restrict_base` (restricting the base of g from B to A), so I removed it (and now we don't have to worry about naming it).